changelogs/unreleased/move-validate-type-method-to-core.yml

```diff
@@ -0,0 +1,8 @@
+---
+description: Moved the `validate_type` logic from the std module to inmanta-core.
+issue-nr: 6540
+issue-repo: inmanta-core
+change-type: minor
+destination-branches: [master]
```

Comment:
Once this is merged in, I will create a new PR for the ISO6 branch that also bump the `COMPILER_VERSION` on the iso6 branch.